### PR TITLE
Task/navigator home

### PR DESCRIPTION
**Pull request by kimberllyflorez**

### CODE REVIEW

File: lib/app/home/provider/main_provider.dart [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import 'dart:async';
+
+import 'package:equatable/equatable.dart';
+import 'package:flutter/material.dart';
+import 'package:rxdart/rxdart.dart';
+
+enum WidgetsView {
```
Comment (curregoz) [inline lib/app/home/provider/main_provider.dart:7]: I don't understand the name of this enum.
Why did you call it WidgetView?

File: lib/app/home/provider/main_provider.dart [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        index: index ?? this.index);
+  }
+
+  @override
+  List<Object?> get props => [widgetsView, index];
+}
+
+class MainProvider extends ChangeNotifier {
+  final _state = BehaviorSubject<WidgetHeadState>.seeded(
+      const WidgetHeadState(widgetsView: WidgetsView.home, index: 0));
```
Comment (curregoz) [inline lib/app/home/provider/main_provider.dart:34]: You don't assign value to the seeded value of the stream. The values that you are using are the same of the state constructor

File: lib/app/home/provider/main_provider.dart [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  void changeIndex(int index) {
+    if (index == 0) {
+      _state.add(WidgetHeadState(widgetsView: WidgetsView.home, index: index));
+    } else if (_state.value.index == 1) {
+      _state
+          .add(WidgetHeadState(widgetsView: WidgetsView.tracker, index: index));
+    } else {
+      _state.add(WidgetHeadState(widgetsView: WidgetsView.sport, index: index));
+    }
```
Comment (curregoz) [inline lib/app/home/provider/main_provider.dart:46]:
```dart
WidgetsView view = WidgetsView.home;
if(index == 1){
    view = WidgetsView.tracker;
} else if(index == 2){
   view = WidgetsView.sport;
}
_state.add(WidgetHeadState(widgetsView: view, index: index));
notifyListeners();
```

File: lib/app/home/ui/pages/head_page.dart
Comment (curregoz) [inline lib/app/home/ui/pages/head_page.dart:1]: Why is this file called head_page and the class is called mainPage?

File: lib/app/home/ui/pages/head_page.dart [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                      icon: Icon(Icons.home), label: 'home'),
+                  BottomNavigationBarItem(
+                      icon: Icon(Icons.track_changes_outlined),
+                      label: 'tracker'),
+                  BottomNavigationBarItem(
+                      icon: Icon(Icons.sports_gymnastics), label: 'sport'),
+                ],
+                currentIndex: state?.index ?? 0,
+                selectedItemColor: AppTheme.primary,
+                onTap: (index) => pageProvider.changeIndex(index)),
```
Comment (curregoz) [inline lib/app/home/ui/pages/head_page.dart:59]: Add trailing comma

File: lib/app/home/ui/pages/head_page.dart [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Provider(
+      create: (_) => pageProvider,
+      child: StreamBuilder<WidgetHeadState>(
+        stream: pageProvider.state,
+        builder: (context, snapshot) {
+          final state = snapshot.data;
+          Widget body = const CircularProgressIndicator();
```
Comment (curregoz) [inline lib/app/home/ui/pages/head_page.dart:35]: I think that we can improve this code creating a map with the widget that you need.

```dart
Map<WidgetsView, Widget> body = {
  WidgetsView.tracker: const SearchProductPage(),
    WidgetsView.tracker: const HomePage(),
.....    WelcomePage
}
final body = body[state.widgetsView] ?? HomePage();
body: body
```

Comment (kimberllyflorez) [reply lib/app/home/ui/pages/head_page.dart]: i don't understand how implement this at the same time validate if data is null

**Pull request merged**